src/averageAges.js

```diff
@@ -14,12 +14,16 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = null) {
+  const allMenAge = people
+    .filter(person => century
+      ? (Math.ceil(person.died / 100) === century)
+      && person.sex === 'm'
+      : person.sex === 'm')
+    .map(person => person.died - person.born);
+
+  return Math.round(allMenAge.reduce((acc, age) => acc + age)
+  / allMenAge.length * 100) / 100;
```

Comment:
you repeat this action, it`s better to create function

Reply:
Done.